psa-crypto/src/types/key.rs

```diff
@@ -428,12 +428,41 @@ impl Attributes {
         })
     }
 
-    /// Sufficient buffer size for the MAC of the specified algorithm, if compatible
+    /// Sufficient buffer size for an encrypted message using the given aead algorithm
     #[cfg(feature = "interface")]
-    pub fn mac_length(self, mac_alg: Mac) -> Result<usize> {
-        self.compatible_with_alg(mac_alg.into())?;
+    pub fn aead_encrypt_output_size(self, alg: Aead, plaintext_len: usize) -> Result<usize> {
+        self.compatible_with_alg(alg.into())?;
+        Ok(unsafe {
+            psa_crypto_sys::PSA_AEAD_ENCRYPT_OUTPUT_SIZE(
+                /*self.key_type.try_into() PSA API specifies including this parameter*/
+                alg.into(),
+                plaintext_len,
+            )
+        })
+    }
+
+    /// Sufficient buffer size for an encrypted message using the given aead algorithm
+    #[cfg(feature = "interface")]
+    pub fn aead_decrypt_output_size(self, alg: Aead, ciphertext_len: usize) -> Result<usize> {
+        self.compatible_with_alg(alg.into())?;
         Ok(unsafe {
-            psa_crypto_sys::PSA_MAC_LENGTH(self.key_type.try_into()?, self.bits, mac_alg.into())
+            psa_crypto_sys::PSA_AEAD_DECRYPT_OUTPUT_SIZE(
+                /*self.key_type.try_into() PSA API specifies including this parameter*/
+                alg.into(),
+                ciphertext_len,
+            )
+        })
+    }
+
+    /// Sufficient buffer size for the resulting shared secret from a raw key agreement
+    #[cfg(feature = "interface")]
+    pub fn raw_key_agreement_output_size(self, alg: RawKeyAgreement) -> Result<usize> {
+        self.compatible_with_alg(KeyAgreement::Raw(alg).into())?;
+        Ok(unsafe {
+            psa_crypto_sys::PSA_RAW_ECDH_KEY_AGREEMENT_OUTPUT_SIZE(
```

Comment:
Should we check that `alg` Is only ECDH?

Reply:
Yes 👀